Add note about immutable release tags (#1950)
Azure/azure-sdk: docs/policies/repobranching.md

Format of the tag should be `<package-name>_<package-version>`

> **_Note:_** Our release tags should be considered immutable and we should avoid updating or deleting them after we have pushed them. If you need to update or delete one for some exceptional case please reach out the engineering system team to discuss options. 

Reviewer: ```suggestion
**_Note:_** Our release tags should be considered immutable.  Avoid updating or deleting them after they are pushed. If you need to update or delete one for some exceptional case, please contact the engineering system team to discuss options. 
```

> **_Note:_** Our release tags should be considered immutable and we should avoid updating or deleting them after we have pushed them. If you need to update or delete one for some exceptional case please reach out the engineering system team to discuss options. 

Reviewer 2: Should we also include `GitHub Releases`, and not just `tags` in this note?

Author: Personally I don't think we are getting much value out of our github releases as the only place to view them is on github and since we have so many it is difficult the find the one you are looking for and so unless someone has a direct link to a release (which I'm not sure they exist in any common location) they are kind of useless. I've actually considered whether or not we should stop creating them, but for now I've just been letting them go. For that reason I've not added any mention of them here.

Reviewer 2: Makes sense.

Format of the tag should be `<package-name>_<package-version>`

> **_Note:_** Our release tags should be considered immutable.  Avoid updating or deleting them after they are pushed. If you need to update or delete one for some exceptional case, please contact the engineering system team to discuss options. 

Reviewer 2: inconsistency nit: two spaces after period
```suggestion
**_Note:_** Our release tags should be considered immutable. Avoid updating or deleting them after they are pushed. If you need to update or delete one for some exceptional case, please contact the engineering system team to discuss options. 
```
